src/libs/SearchParser/searchParser.peggy

```diff
@@ -36,93 +38,102 @@
 
+  return allFilters.reduce((result, filter) => buildFilter("and", result, filter));
+}
+    
 filter
-  = _ field:key? _ op:operator? _ value:identifier {
-      if (isDefaultField(field)) {
-        updateDefaultValues(field, value.trim());
-        return null;
-      }
+= @(standardFilter / defaultFilter / multiSelectFilter / freeTextFilter )
+
+defaultFilter
+= _ key:defaultKey _ op:operator _ value:identifier {
+	updateDefaultValues(key, value.trim());
+}
 
-      if (!field && !op) {
-        return buildFilter('eq', 'keyword', value.trim());
-      }
+freeTextFilter
+= _ value:identifier _ {
+	return buildFilter('eq', 'keyword', value.trim());
+}
 
-      const values = value.split(',');
-      const operatorValue = op ?? 'eq';
+multiSelectFilter
+= _ field:multiselectKey _ op:operator _ value:identifier {
+  return buildFilter(op, field, value)
+}
 
-      return values.slice(1).reduce((acc, val) => buildFilter('or', acc, buildFilter(operatorValue, field, val.trim())), buildFilter(operatorValue, field, values[0]));
-    }
+standardFilter
+= _ field:key _ op:operator _ value:identifier {
+	
+	const values = value.split(',');
+    return values.slice(1).reduce((acc, val) => buildFilter('or', acc, buildFilter(op, field, val.trim())), buildFilter(op, field, values[0]));
+}
 
 operator
   = (":" / "=") { return "eq"; }
   / "!=" { return "neq"; }
-  / ">" { return "gt"; }
   / ">=" { return "gte"; }
-  / "<" { return "lt"; }
+  / ">" { return "gt"; }
   / "<=" { return "lte"; }
+  / "<" { return "lt"; }
 
-key
-  = "type" { return "type"; }
-  / "status" { return "status"; }
-  / "date" { return "date"; }
-  / "amount" { return "amount"; }
-  / "expenseType" { return "expenseType"; }
-  / "in" { return "in"; }
-  / "currency" { return "currency"; }
-  / "merchant" { return "merchant"; }
-  / "description" { return "description"; }
-  / "from" { return "from"; }
-  / "to" { return "to"; }
-  / "category" { return "category"; }
-  / "tag" { return "tag"; }
-  / "taxRate" { return "taxRate"; }
-  / "cardID" { return "cardID"; }
-  / "reportID" { return "reportID"; }
-  / "keyword" { return "keyword"; }
-  / "sortBy" { return "sortBy"; }
-  / "sortOrder" { return "sortOrder"; }
+key "key"
+  = @("date" 
+  / "amount" 
+  / "expenseType"
+  / "merchant" 
+  / "description" 
+  / "reportID"
+  / "keyword")
+  
+defaultKey "default key"
+= @("type"
+  / "status"
+  / "sortBy"
+  / "sortOrder")
+  
+multiselectKey "multiselect key"
+= @("in"
+  / "currency"
+  / "tag"
+  / "category"
+  / "to"
+  / "taxRate"
+  / "cardID"
+  / "from")
 
 identifier
  = parts:(quotedString / alphanumeric)+ { return parts.join(''); }
 
-quotedString
+quotedString "quote"
   = '"' chars:[^"\r\n]* '"' { return chars.join(''); }
 
-alphanumeric
+alphanumeric "word"
   = chars:[A-Za-z0-9_@./#&+\-\\',;]+ { return chars.join(''); }
 
-logicalAnd
+logicalAnd "and"
   = _ { return "and"; }
 
 _ "whitespace"
-  = [ \t\r\n]*
+  = [ \t\r\n]*
```

Comment:
for every lacking newline, one penguin is dying somewhere

Reply:
to be precise a _baby penguin_